app-service-deploy/entrypoint.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -e
+
+LINUX_APP_SUBSTRING='linux'
+
+if [[ -z $WEB_APP_NAME ]];
+then 
+    echo "Required Web App Name. Provide value in WEB_APP_NAME variable" >&2
+    exit 1
+fi
+echo "Web App Name: ${WEB_APP_NAME}"
+
+if [[ -z $RESOURCE_GROUP_NAME ]];
+then 
+    RESOURCE_GROUP_NAME=`az resource list -n "${WEB_APP_NAME}" --resource-type "Microsoft.Web/Sites" --query '[0].resourceGroup' | xargs`
+    if [[ -z $RESOURCE_GROUP_NAME ]];
+    then
+        echo "Web App '${WEB_APP_NAME}' should exist before deployment." >&2
+        exit 1
+    fi
+fi
+
+echo "Resource Group Name: ${RESOURCE_GROUP_NAME}"
+
+APP_KIND=`az resource show -n "${WEB_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" --resource-type "Microsoft.Web/Sites" --query 'kind'` 
+echo "Web App type : ${APP_KIND}"
+
+echo "Initiating Web App Deployment"
+if [[ ! $APP_KIND =~ $LINUX_APP_SUBSTRING ]];
+then
+    echo "Setting App Setting WEBSITE_RUN_FROM_PACKAGE = 1 ..."
+    az webapp config appsettings set -g "${RESOURCE_GROUP_NAME}" -n "${WEB_APP_NAME}" --settings WEBSITE_RUN_FROM_PACKAGE=1 
+    sleep 10 # TODO: find whether this app setting is updated in Kudu
+    echo "Set WEBSITE_RUN_FROM_PACKAGE = 1 successfully!"
+fi
+
+if [[ -z $PACKAGE_PATH ]];
+then
+    PACKAGE_PATH=$GITHUB_WORKSPACE
+fi
+
+echo "Provided package path ${PACKAGE_PATH}"
+
+if [[ -d $PACKAGE_PATH ]];
+then
+    echo "Compressing Package '${PACKAGE_PATH}' to '$GITHUB_WORKSPACE/PACKAGE.zip'"
+    cd $PACKAGE_PATH
+    zip -r "$GITHUB_WORKSPACE/PACKAGE.zip" * 
+    PACKAGE_PATH="$GITHUB_WORKSPACE/PACKAGE.zip"
```

Comment:
do you want to move this statement one step ahead and use zip -r $PACKAGE_PATH

Reply:
it will zip with current folder as well. so we cd into package root and start the zip process there.